relay-sampling/src/evaluation.rs

```diff
@@ -195,19 +195,21 @@ impl<'a> SamplingEvaluator<'a> {
         I: Iterator<Item = &'b SamplingRule>,
     {
         for rule in rules {
-            if !rule.condition.matches(instance) {
+            if !rule.time_range.contains(self.now) || !rule.condition.matches(instance) {
                 continue;
             };
 
-            let Some(sampling_value) = rule.evaluate(self.now, self.reservoir) else {
+            let Some(sampling_value) = rule.adjusted_sampling_value(self.now) else {
```

Comment:
at this point, it is no longer determined if there is a match or not, as it is only determined in the following match statement in the case of the reservoir variant.

Perhaps that's considered bad design, it's hard to make it ideal within the current schema.